vignettes/gooey_edge/lib/gooey_carousel.dart

```diff
@@ -41,42 +40,47 @@ class GooeyCarouselState extends State<GooeyCarousel> with SingleTickerProviderS
 
   void _tick(Duration duration) {
     _edge.tick(duration);
-    setState((){});
+    setState(() {});
   }
 
   @override
   Widget build(BuildContext context) {
     int l = widget.children.length;
 
     return GestureDetector(
-      key: _key,
-      onPanDown: (details) => _handlePanDown(details, _getSize()),
-      onPanUpdate: (details) => _handlePanUpdate(details, _getSize()),
-      onPanEnd: (details) => _handlePanEnd(details, _getSize()),
-
-      child: Stack(children: <Widget>[
-        widget.children[_index % l],
-        
-        _dragIndex == null ? SizedBox() : ClipPath(
-          child: widget.children[_dragIndex % l],
-          clipBehavior: Clip.hardEdge,
-          clipper: GooeyEdgeClipper(_edge, margin: 10.0),
-        ),
-        SunAndMoon(
-          index: _dragIndex,
-          isDragComplete: _dragCompleted,
-          )
-      ],)
-    );
+        key: _key,
+        onPanDown: (details) => _handlePanDown(details, _getSize()),
+        onPanUpdate: (details) => _handlePanUpdate(details, _getSize()),
+        onPanEnd: (details) => _handlePanEnd(details, _getSize()),
+        child: Stack(
+          children: <Widget>[
+            widget.children[_index % l],
+            _dragIndex == null
+                ? SizedBox()
+                : ClipPath(
+                    child: widget.children[_dragIndex! % l],
+                    clipBehavior: Clip.hardEdge,
+                    clipper: GooeyEdgeClipper(_edge, margin: 10.0),
+                  ),
+            SunAndMoon(
+              //TODO: check this one. is ?? 0 ok?
+              index: _dragIndex ?? 0,
+              isDragComplete: _dragCompleted,
+            )
+          ],
+        ));
   }
 
   Size _getSize() {
-    final RenderBox box = _key.currentContext.findRenderObject();
-    return box.size;
+    final RenderBox? box = _key.currentContext?.findRenderObject() as RenderBox;
```

Comment:
Never cast away nullability.  `findRenderObject` returns `RenderObject?` so it must be cast as `RenderBox?` not `RenderBox`